week7/vm.h

```diff
@@ -0,0 +1,24 @@
+#ifndef VENDINGMACHINE_H_
+#define VENDINGMACHINE_H_
+
+#include "inventory.h"
+
+class VendingMachine {
+public:
+  VendingMachine(double balance, Inventory inventory);
+  virtual ~VendingMachine();
+
+  void cancel();
+  void recvMoney(int amount);
+  int selectItem(int itemID);
+  void reset();
+
+private:
+  double totalBalance;
+  Inventory inventory;
+
+  void issueRefund();
+  double getTotalBalance();
+};
+
+#endif /* VENDINGMACHINE_H_ */
```

Comment:
Thanks for uploading your hand-written notes! You had good test cases. I liked that you were thinking about how the machine would be used and what its behavior needed to look like. You're correct about the TEBOW IT not fitting as nicely into design problems. A lot of the planning is better served with UML diagrams or something similar.

Feel free to ping me if you get back to the problem and I'll be happy to provide feedback!